crates/interpreter/src/instructions/i256.rs

```diff
@@ -31,102 +30,101 @@ pub const MIN_NEGATIVE_VALUE: U256 = U256::from_limbs([
-    if !val.bit(U256::BITS - 1) {
-        if *val == U256::ZERO {
-            Sign::Zero
-        } else {
-            Sign::Plus
-        }
-    } else {
-        if DO_TWO_COMPL {
-            two_compl_mut(val);
-        }
+pub(super) fn i256_sign(val: &U256) -> Sign {
+    if val.bit(U256::BITS - 1) {
         Sign::Minus
+    } else {
+        // SAFETY: false == 0 == Zero, true == 1 == Plus
+        unsafe { core::mem::transmute(*val != U256::ZERO) }
     }
 }
 
+#[inline(always)]
+pub(super) fn i256_sign_compl(val: &mut U256) -> Sign {
+    let sign = i256_sign(val);
+    if sign == Sign::Minus {
+        two_compl_mut(val);
+    }
+    sign
+}
+
 #[inline(always)]
 fn u256_remove_sign(val: &mut U256) {
+    // SAFETY: U256 does not have any padding bytes
     unsafe {
         val.as_limbs_mut()[3] &= FLIPH_BITMASK_U64;
     }
 }
 
 #[inline(always)]
-pub fn two_compl_mut(op: &mut U256) {
+pub(super) fn two_compl_mut(op: &mut U256) {
     *op = two_compl(*op);
 }
 
-pub fn two_compl(op: U256) -> U256 {
+#[inline(always)]
+pub(super) fn two_compl(op: U256) -> U256 {
     op.wrapping_neg()
 }
 
 #[inline(always)]
-pub fn i256_cmp(mut first: U256, mut second: U256) -> Ordering {
-    let first_sign = i256_sign::<false>(&mut first);
-    let second_sign = i256_sign::<false>(&mut second);
-    match (first_sign, second_sign) {
-        (Sign::Zero, Sign::Zero) => Ordering::Equal,
-        (Sign::Zero, Sign::Plus) => Ordering::Less,
-        (Sign::Zero, Sign::Minus) => Ordering::Greater,
-        (Sign::Minus, Sign::Zero) => Ordering::Less,
-        (Sign::Minus, Sign::Plus) => Ordering::Less,
-        (Sign::Minus, Sign::Minus) => first.cmp(&second),
-        (Sign::Plus, Sign::Minus) => Ordering::Greater,
-        (Sign::Plus, Sign::Zero) => Ordering::Greater,
-        (Sign::Plus, Sign::Plus) => first.cmp(&second),
+pub(super) fn i256_cmp(first: &U256, second: &U256) -> Ordering {
+    let first_sign = i256_sign(first);
+    let second_sign = i256_sign(second);
+    match first_sign.cmp(&second_sign) {
+        // note: adding `if first_sign != Sign::Zero` to short circuit zero comparisons performs
+        // slower on average, as of #582
+        Ordering::Equal => first.cmp(second),
+        o => o,
     }
 }
 
 #[inline(always)]
-pub fn i256_div(mut first: U256, mut second: U256) -> U256 {
-    let second_sign = i256_sign::<true>(&mut second);
+pub(super) fn i256_div(mut first: U256, mut second: U256) -> U256 {
+    let second_sign = i256_sign_compl(&mut second);
     if second_sign == Sign::Zero {
         return U256::ZERO;
     }
-    let first_sign = i256_sign::<true>(&mut first);
+    let first_sign = i256_sign_compl(&mut first);
     if first_sign == Sign::Minus && first == MIN_NEGATIVE_VALUE && second == U256::from(1) {
         return two_compl(MIN_NEGATIVE_VALUE);
     }
 
-    //let mut d = first / second;
-    let mut d = first.div_rem(second).0;
+    let mut d = first.wrapping_div(second);
```

Comment:
`wrapping_div` calls `div_rem` in background, we can even do `first / second` as it calls same things